packages/common-scripts/src/omnilock-bitcoin.ts

```diff
@@ -1,43 +1,56 @@
-// TODO the magic number eslint will be resolved in 0.24 by recovering https://github.com/ckb-js/lumos/pull/682
-/*eslint-disable @typescript-eslint/no-magic-numbers*/
-
 import { bytes, BytesLike } from "@ckb-lumos/codec";
 import { bech32 } from "bech32";
 import bs58 from "bs58";
 
+export type SupportedBtcAddressType = "P2SH-P2WPKH" | "P2WPKH" | "P2PKH";
+
 // https://github.com/cryptape/omnilock/blob/9419b7795641da0ade25a04127e25d8a0b709077/c/ckb_identity.h#L28
 const BTC_PREFIX = "CKB (Bitcoin Layer) transaction: 0x";
 
 /**
  * Decode bitcoin address to public key hash in bytes
  * @see https://en.bitcoin.it/wiki/List_of_address_prefixes
  * @param address
+ * @param allows
  */
-export function decodeAddress(address: string): ArrayLike<number> {
-  try {
-    // Bech32
-    if (address.startsWith("bc1q")) {
-      return bech32.fromWords(bech32.decode(address).words.slice(1));
-    }
+export function decodeAddress(
+  address: string,
+  allows: SupportedBtcAddressType[]
+): ArrayLike<number> {
+  const btcAddressFlagSize = 1;
+  const hashSize = 20;
 
-    // P2PKH
-    if (address.startsWith("1")) {
-      return bs58.decode(address).slice(1, 21);
-    }
+  if (isP2wpkhAddress(address) && allows.includes("P2WPKH")) {
+    return bech32.fromWords(bech32.decode(address).words.slice(1));
```

Comment:
`allows.includes("P2WPKH")` should be checked inside this branch and throw directly if `P2WPKH` is not allowed.

Reply:
refactored